src/mutations/member.ts

```diff
@@ -88,6 +88,46 @@ export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
     },
   });
 
+  // suppose you can only edit yourself
+  queryClient.setMutationDefaults(MUTATION_KEYS.UPDATE_PASSWORD, {
+    mutationFn: (payload) =>
+      Api.updatePassword(payload, queryConfig).then((member) => Map(member)),
+    onMutate: async ({ member }) => {
+      // Cancel any outgoing refetches (so they don't overwrite our optimistic update)
+      await queryClient.cancelQueries(CURRENT_MEMBER_KEY);
+
+      // Snapshot the previous value
+      const previousMember = queryClient.getQueryData(
+        CURRENT_MEMBER_KEY,
+      ) as Record<Member>;
+
+      // Optimistically update to the new value
+      queryClient.setQueryData(
+        CURRENT_MEMBER_KEY,
+        previousMember.merge(member),
+      );
+
+      // Return a context object with the snapshotted value
+      return { previousMember };
+    },
+    onSuccess: () => {
+      notifier?.({
+        type: editMemberRoutine.SUCCESS,
+        payload: { message: SUCCESS_MESSAGES.EDIT_MEMBER },
```

Comment:
I think you can create new routine especially for updating the password (it will allow you to have a particular notification message as well for this operation).

Reply:
I am not able to get the routine working the correct way. Anyway, the SUCCESS notification messages can be set manually and the FAILURE messages are returned by the core